extensions/amp-samba-player/0.1/amp-samba-player.js

```diff
@@ -0,0 +1,260 @@
+
+    /** @private {Object} */
+    this.params_ = {};
+
+    /** @private {?HTMLIFrameElement} */
+    this.iframe_ = null;
+
+    /** @private {?Function} */
+    this.playerLoadResolver_ = null;
+
+    /** @private {Promise} */
+    this.playerLoadPromise_ = new Promise(resolve => {
+      this.playerLoadResolver_ = resolve;
+    });
+
+    // Event handlers bindings
+    this.playerMessageHandler_ = this.playerMessageHandler_.bind(this);
+  }
+
+  /** @override */
+  preconnectCallback(opt_onLayout) {
+    // host to serve the player
+    this.preconnect.url(API_MAP['prod'], opt_onLayout);
+    // host to serve media contents
+    this.preconnect.url('http://pvbps-sambavideos.akamaized.net', opt_onLayout);
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    // support all layouts
+    return isLayoutSizeDefined(layout);
+  }
+
+  /** @override */
+  buildCallback() {
+    this.projectId_ = user().assert(
+      this.element.getAttribute('data-project-id'),
+      `The data-project-id attribute is required for <${TAG}> %s`,
+      this.element);
+
+    // not required in case of live
+    this.mediaId_ = this.element.getAttribute('data-media-id') || '';
+
+    // which environment to run
+    this.env_ = this.element.getAttribute('data-env') || this.env_;
+
+    // the protocol to be used (HTTP, HTTPS)
+    this.protocol_ = (this.element.getAttribute('data-protocol')
+      || this.protocol_).replace(/\:?$/, ':');
+
+    // player features related params
+    // WARN: methods missing on returned object (e.g. hasOwnProperty) so recreate it
+    this.params_ = Object.assign({}, getDataParamsFromAttributes(this.element));
+
+    // remove auto-start attribute (video manager will take care of it)
+    if ('autoStart' in this.params_) {
+      delete this.params_['autoStart'];
+      user().error(TAG, 'Use autoplay attribute instead of '
+        + 'data-param-auto-start.');
+    }
+
+    installVideoManagerForDoc(this.element);
+    videoManagerForDoc(this.element).register(this);
+  }
+
+  /** @override */
+  layoutCallback() {
+    const iframe = this.element.ownerDocument.createElement('iframe');
+    let params = 'jsApi=true';
+    const s = '&';
+
+    for (const k in this.params_) {
+      params += `${s}${k.replace(/url$/i, 'URL')}=${this.params_[k]}`;
+    }
+
+    params += `&parentURL=#${this.element.ownerDocument.location.href}`;
+
+    iframe.setAttribute('frameborder', '0');
+    iframe.setAttribute('allowfullscreen', 'true');
+    iframe.src = `${this.protocol_}${API_MAP[this.env_]}${this.projectId_}/`
+      + `${this.mediaId_}?${params}`;
+
+    this.applyFillContent(iframe);
+    this.element.appendChild(iframe);
+    this.iframe_ = iframe;
+
+    this.win.addEventListener('message', this.playerMessageHandler_);
+
+    return this.loadPromise(iframe)
+      .then(() => {
+        this.element.dispatchCustomEvent(VideoEvents.LOAD);
+        this.playerLoadResolver_(this.iframe_);
```

Comment:
this can jsut become

```
this.playerLoadPromise_ = this.loadPromise(iframe)
      .then(() => {
         this.element.dispatchCustomEvent(VideoEvents.LOAD);
      });
return  this.playerLoadPromise_;
```